src/modals/SendModal.js

```diff
@@ -283,6 +284,24 @@ class SendModal extends Component {
     this.props.sendModalInit();
   };
 
+  updateEns = target => {
+    if (target.value.includes('.')) {
+      ens
+        .lookup(target.value.trim())
+        .then(address => {
+          this.props.sendUpdateRecipient(address, this.props.selected.symbol);
+        })
+        .catch(reason => {
+          console.log('ENS failed');
+          // console.error(reason);
+        });
+    }
+    this.props.sendUpdateRecipient(
+      target.value.trim(),
+      this.props.selected.symbol,
+    );
+  };
+
```

Comment:
The problem with this logic is that it still updates the recipient input which is in focus

This logic should move to the `_send.js` reducer and handled a state variable for input which will then update the recipient state with the result of the ens lookup

The target.value.includes('.') should check for '.eth' instead and there should a timeout of roughly 300ms to allow the user to allow the user to finish typing

Reply:
Thanks for the review, makes sense, shifting the logic. :+1:

Reply:
Can you please recheck, since the input value for me isn't updated - 
![image](https://user-images.githubusercontent.com/21009455/41083176-263b7090-6a4e-11e8-8521-52f6c4ec9284.png)

Reply:
Send button is disabled when ENS resolution fails.
![image](https://user-images.githubusercontent.com/21009455/41083235-4aa2e24c-6a4e-11e8-94fd-348111663fdb.png)

Reply:
@pedrouid ping.

Reply:
@ricburton, please see :sweat_smile: